File: analytics/src/main/java/io/appmetrica/analytics/impl/component/clients/ClientUnit.java
package io.appmetrica.analytics.impl.component.clients;

import androidx.annotation.NonNull;
import io.appmetrica.analytics.impl.CounterReport;
import io.appmetrica.analytics.impl.component.CommonArguments;

public interface ClientUnit {

    void handle(@NonNull CounterReport report, @NonNull CommonArguments sdkConfig);

    void onDisconnect();

}

File: analytics/src/main/java/io/appmetrica/analytics/impl/component/clients/ClientUnit.kt
package io.appmetrica.analytics.impl.component.clients

import io.appmetrica.analytics.impl.CoreServiceEvent
import io.appmetrica.analytics.impl.component.CommonArguments

internal interface ClientUnit {
    fun handle(serviceEvent: CoreServiceEvent, sdkConfig: CommonArguments)

    fun onDisconnect()
}
